server/coordinator/procedure/manager_impl.go

```diff
@@ -123,34 +122,62 @@ func (m *ManagerImpl) retryAll(ctx context.Context) error {
 	return nil
 }
 
-func (m *ManagerImpl) startProcedureWorker(ctx context.Context, procedures <-chan Procedure) {
-	for procedure := range procedures {
-		err := procedure.Start(ctx)
-		if err != nil {
-			log.Error("procedure start failed", zap.Error(err))
+func (m *ManagerImpl) startProcedurePromote(ctx context.Context) {
+	ticker := time.NewTicker(defaultPromoteDelay)
+	procedureWorkerChan := make(chan bool)
```

Comment:
- How about making it a buffered channel?
- And why not let it be a field of the `ManagerImpl` so that it can be notified when submitting a procedure?
- Replace `chan bool` to `chan struct{}`.